src/models/Settings.js

```diff
@@ -223,19 +227,55 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
         },
         cache: true
       },
-      // checks if there are any valid configured idps.
-      socialAuthConfigured: {
-        deps: ['configuredSocialIdps'],
-        fn: function (idps) {
-          return !_.isEmpty(idps);
+      // social auth buttons order - 'above'/'below' the primary auth form (boolean)
+      socialAuthPositionTop: {
+        deps: ['configuredSocialIdps', 'idpDisplay'],
+        fn: function (configuredSocialIdps, idpDisplay) {
+          return !!(!_.isEmpty(configuredSocialIdps) && idpDisplay.toUpperCase() === 'PRIMARY');
```

Comment:
nit: not sure if you need !! at the beginning. Both conditions return a boolean...